styx-common/src/test/java/com/spotify/styx/util/BasicWorkflowValidatorTest.java

```diff
@@ -267,6 +275,118 @@ public void shouldAcceptValidServiceAccount(String serviceAccount) {
             empty());
   }
 
+  @Test
+  public void shouldAcceptValidDockerExecConf() {
+    assertThat(
+        sut.validateWorkflow(Workflow.create("test", DOCKER_EXEC_WORKFLOW_CONFIGURATION)),
+        empty()
+    );
+  }
+
+  @Test
+  public void shouldRejectInvalidImageInDockerExecConf() {
+    when(dockerImageValidator.validateImageReference(anyString())).thenReturn(Set.of("foo", "bar"));
+
+    var errors = sut.validateWorkflow(Workflow.create("test", DOCKER_EXEC_WORKFLOW_CONFIGURATION));
```

Comment:
Should we instead create an INVALID_DOCKER_EXEC_WORKFLOW_CONFIGURATION, and make this test a bit more explicit?